src/cmds/core/ban.py

```diff
@@ -24,6 +24,49 @@ class BanCog(commands.Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    class BanApprove(discord.ui.view):
```

Comment:
`discord.ui.view` is not a class, so you cannot extend. Maybe you wanted to write `discord.ui.view.View`. In that case better import it `from discord.ui.view import View` and use it.

Reply:
Also, why BanApprove is a nested class? This is uncommon pattern in Python.